_posts/en/newsletters/2025-01-03-newsletter.md

```diff
@@ -0,0 +1,309 @@
+candidates, and describing notable changes to popular Bitcoin
+infrastructure software.
+
+## News
+
+- **Deanonymization attacks against centralized coinjoin:** Yuval Kogman
+  [posted][kogman cc] to the Bitcoin-Dev mailing list details about several
+  privacy-reducing vulnerabilities in the centralized [coinjoin][topic
+  coinjoin] protocols used by current versions of the Wasabi and Ginger
+  wallets, plus past versions of the Samourai, Sparrow, and Trezor Suite
+  software wallets.  Kogman helped design the WabiSabi protocol used in
+  Wasabi and Ginger (see [Newsletter #102][news102 wabisabi]) but "left
+  in protest before its release." If exploited, the vulnerabilities
+  allow the centralized coordinator to determine which user received
+  which outputs, eliminating any benefit of using a sophisticated
+  protocol over a simple webserver.  Kogman provides evidence that
+  the vulnerabilities were known to several wallet developers for years.
+  A similar vulnerability affecting some of the same software was
+  previously mentioned in [Newsletter #333][news333 vuln].
+
+- **Updated ChillDKG draft:** Tim Ruffing and Jonas Nick [posted][rn
+  chilldkg] to the Bitcoin-Dev mailing list a link to the [current draft
+  BIP for ChillDKG][bip-chilldkg], which describes a distributed key
+  generation protocol compatible with FROST [scriptless threshold
+  signatures][topic threshold signature] for Bitcoin.  Since their
+  initial announcement (see [Newsletter #312][news312 chilldkg]), they've
+  fixed a security vulnerability, added an investigation phase that
+  makes it possible to identify faulty participants, and made backup and
+  recovery easier.  They've also worked with Sivaram Dhakshinamoorthy to
+  keep their proposal in sync with his proposal for Bitcoin-compatible
+  FROST signing (see [Newsletter #315][news315 frost]).
+
+## Changing consensus
+
+_A new monthly section summarizing proposals and discussion about
+changing Bitcoin's consensus rules._
+
+- **CTV enhancement opcodes:** developer moonsettler posted to both
+  [Bitcoin-Dev][moonsettler ctvppml] and [Delving Bitcoin][moonsettler
+  ctvppdelv] to propose two additional opcodes to be used in conjunction
+  with the already-proposed [OP_CHECKTEMPLATEVERIFY][topic
+  op_checktemplateverify]:
+
+  - _OP_TEMPLATEHASH_ takes a list of transaction elements and converts
+    it into a CTV-compatible hash.  This allows stack operations to
+    specify details about the inputs to be used, the number of inputs,
+    the locktime, the value of each output, the script of each output,
+    the number of outputs, and the transaction version to use.
+
+  - _OP_INPUTAMOUNTS_ puts the satoshi value of some or all of the
+    inputs on the stack, which can be used as a parameter for
+    `OP_TEMPLATEHASH` (e.g. to require an equal output value).
+
+  Together these opcodes can create [vaults][topic vaults] with similar
+  properties to those possible with [BIP345][]'s `OP_VAULT`.  The
```

Comment:
Not quite. While this does allow spending multiple vaulted outputs together, it is missing two of `OP_VAULT` properties:

- No partial re-vaulting support, the vaulted outputs must be spent in full to the staging script.
- No dynamic construction of the destination tap leaf, which enables choosing the withdrawal destination address at trigger time. Vaults using moonsettler's proposal would be limited to using static pre-determined destination addresses, similarly to simple `OP_CTV`-only vaults.